include/tdi/common/tdi_table_key.hpp

```diff
@@ -66,30 +90,71 @@ class KeyFieldValueTernary : public KeyFieldValue {
  public:
   KeyFieldValueTernary(T &value, T &mask)
       : value_(value), mask_(mask){};
-  KeyFieldValueTernary(uint8_t *value_ptr,
-                       uint8_t *mask_ptr,
-                       size_t &size)
+  KeyFieldValueTernary(T *value_ptr,
+                       T *mask_ptr,
+                       const size_t size)
       : value_ptr_(value_ptr), mask_ptr_(mask_ptr), size_(size){};
   T value_ = 0;
-  uint8_t *value_ptr_ = nullptr;
+  T *value_ptr_ = nullptr;
   T mask_ = 0;
-  uint8_t *mask_ptr_ = nullptr;
+  T *mask_ptr_ = nullptr;
+  size_t size_ = 0;
+};
+template<class T>
+class KeyFieldValueString: public KeyFieldValue {
```

Comment:
String is not a match type. It is a data type. It needs to be one of the templates of exact. Currently Only exact match supports strings.
We need to instatntiate explicit templates for string
`KeyFieldValueExact<std::string>`